app/bucket_list.py

```diff
@@ -0,0 +1,198 @@
+    """ Creates endpoints for BucketListAPI """
+
+    decorators = [auth_user.login_required]
+
+    def __init__(self):
+        self.reqparse = reqparse.RequestParser()
+        self.reqparse.add_argument('bucketlist_name', type=str, \
+        required=True, help='No BucketList name provided', location='json')
+        super(BucketListAPI, self).__init__()
+
+    def get(self, id=None):
+        """ List all the created bucket lists and single ones too """
+        if id:
+            this_bucket_list = BucketList.query.filter_by(bucketlist_id=id, \
+            created_by=g.user.user_id).first()
+            return marshal(this_bucket_list, bucket_list_fields), 200
+
+        else:
+
+            self.reqparse = reqparse.RequestParser()
+            self.reqparse.add_argument('q', type=str, location='args')
+            self.reqparse.add_argument('limit', type=int, location='args', default=20)
+            self.reqparse.add_argument('page', type=int, location='args', default=1)
+
+            args = self.reqparse.parse_args()
+            query = args['q']
+            limit = args['limit']
+            page = args['page']
+
+            if query:
+                bucketlists = BucketList.query.filter(BucketList.bucketlist_name.like('%'+query+'%'), \
+                BucketList.created_by == g.user.user_id).paginate(page, limit, False)
+            else:
+                bucketlists = BucketList.query.filter_by(created_by=g.user.user_id).paginate(page, limit, False)
+
+            if not bucketlists:
+                return {'message': 'bucketlists not found'}, 404
+
+            if bucketlists.has_prev:
+                prev_page = request.url + '?page=' + str(page - 1) + '&limit=' + str(limit)
+            else:
+                prev_page = 'None'
+
+            if bucketlists.has_next:
+                next_page = request.url + '?page=' + str(page + 1) + '&limit=' + str(limit)
+            else:
+                next_page = 'None'
+
+            return {
+                'message': {
+                    'next_page':next_page,
+                    'prev_page': prev_page,
+                    'total': bucketlists.pages},
+                'bucketlists': marshal(bucketlists.items, bucket_list_fields)
+            }, 200
+
+
+    def post(self):
+        """ Create a new bucket list """
+        args = self.reqparse.parse_args()
+        name = args['bucketlist_name']
+
+        check_duplicate = BucketList.query.filter_by(bucketlist_name=name).first()
+
+        if check_duplicate:
+            return {'error': 'bucketlist_name {} already exists'.format(name)}
+
+        new_bucketlist = BucketList(bucketlist_name=name, created_by=g.user.user_id)
+        db.session.add(new_bucketlist)
+        db.session.commit()
+
+        return marshal(new_bucketlist, bucket_list_fields)
+
+    def put(self, id):
+        """ Update a bucket list """
+        this_bucket_list = BucketList.query.filter_by(bucketlist_id=id).first()
```

Comment:
L118 you're filtering by id only which means you are not considering the the permission aspect here. What if the bucketlist belongs to another user? You might do`this_bucket_list = BucketList.query.filter_by(bucketlist_id=id, created_by=g.user.user_id).first()`